rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -277,14 +277,15 @@ class Subscription : public SubscriptionBase
       return;
     }
     auto typed_message = std::static_pointer_cast<CallbackMessageT>(message);
-    any_callback_.dispatch(typed_message, message_info);
 
     if (subscription_topic_statistics_) {
       const auto nanos = std::chrono::time_point_cast<std::chrono::nanoseconds>(
         std::chrono::system_clock::now());
       const auto time = rclcpp::Time(nanos.time_since_epoch().count());
       subscription_topic_statistics_->handle_message(*typed_message, time);
     }
+
+    any_callback_.dispatch(typed_message, message_info);
```

Comment:
I would not do this because this really changes behavior, instead i would do,

1. if statistics enabled, get current time.
2. call subscription user callback.
3. if statistics enabled, call handle_message with time which is collected on 1

Reply:
Thanks for pointing this out!
I moved the handle_message to after the callback execution, is it as you intended?
I appreciate your comments even small things.